### PR TITLE
Image color select 2: Malen für Zahlen

### LINKED ISSUE #660: Better Field Color Detection Tooling
We should improve our tooling for the field color detection to tune it more easily

### PR DESCRIPTION
**Pull request by pejotejo**
## Why? What?
adds a paint and plot feature to the image colour select panel to hopefully ease vision tuning

Fixes #660 

## Ideas for Next Iterations (Not This PR)
change the vision parameters by dragging their lines in the plot

## How to Test
open the panel and paint field and not field with the two colours (second with shift left click), then look at the plot below

### CODE REVIEW

File: mask_field.png
Comment (knoellle) [inline mask_field.png:1]: You probably want to remove this file

File: tools/twix/src/panels/image_color_select.rs [hunk 1]
```diff
@@ -28,272 +31,393 @@ use crate::{
...
     }
 }
 
-pub struct ImageColorSelectPanel {
-    nao: Arc<Nao>,
-    image_buffer: ImageBuffer,
-    cycler_selector: VisionCyclerSelector,
-    brush_size: f32,
+#[derive(Default)]
+struct Statistics {
```
Comment (knoellle) [inline tools/twix/src/panels/image_color_select.rs:92]: This type is unused and so is the `ColorArray` above, but clippy isn't smart enough to figure this out.

File: mask_field.png
Comment (pejotejo) [reply mask_field.png:1]: done

File: tools/twix/src/panels/image_color_select.rs (hunk 1 above)
Comment (pejotejo) [reply tools/twix/src/panels/image_color_select.rs]: done, also removed ImageKind

**Comment by knoellle** [comment]
Don't forget to bump the twix version 😉

**Review by h3ndrk — COMMENTED** [review]
Wörks, one small remark :muscle:

File: tools/twix/src/panels/image_color_select.rs [hunk 2]
```diff
@@ -307,8 +376,84 @@ impl<'a> ImageColorSelectPanel {
...
+                let red_chromaticity = rgb.get_chromaticity(RgbChannel::Red);
+                let green_chromaticity = rgb.get_chromaticity(RgbChannel::Green);
+                let blue_chromaticity = rgb.get_chromaticity(RgbChannel::Blue);
+
+                if filter_by_other_axes
+                    && (red_chromaticity > 0.37 && !skip.contains(&Axis::RedChromaticity)
+                        || blue_chromaticity > 0.38 && !skip.contains(&Axis::BlueChromaticity)
+                        || green_chromaticity < 0.5 && !skip.contains(&Axis::GreenChromaticity)
+                        || (rgb.g as f32) < 25.0 && !skip.contains(&Axis::GreenLuminance))
+                    && (rgb.get_luminance() as f32 > 25.0 || skip.contains(&Axis::Luminance))
```
[the remark this answers is not in the document]
Comment (knoellle) [reply tools/twix/src/panels/image_color_select.rs]: Good point, those are copied from our field color config.
Now using the actual field color parameters retrieved from the robot.

**Comment by schmidma** [comment]
What is the state here? This may conflict with my communication rewrite. Should I expect this to be merged soon?

**Comment by knoellle** [comment]
AFAIK this is waiting for #1078 so massimiliano won't have to retrofit his changes onto this.

**Comment by schmidma** [comment]
#1078 was merged 4 days ago. Can you please rebase and resolve conflicts?

**Comment by MaikRe** [comment]
> Don't forget to bump the twix version 😉

This is still missing 😉

**Review by schmidma — APPROVED** [review]

**Pull request merged**